Rename process to handle

gregw opened the pull request:
+ Request.Processor to Request.Handler
 + Handler.process to Handler.handle

gregw committed:
Rename process to handle

 + Request.Processor to Request.Handler
 + Handler.process to Handler.handle

lorban commented on jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebXmlConfiguration.java:
```diff
@@ -131,7 +131,7 @@ public void deconfigure(WebAppContext context) throws Exception
         context.setWelcomeFiles(null);
 
         //TODO: ErrorPageErrorHandler is not an ErrorProcessor
-        if (context.getErrorProcessor() instanceof ErrorPageErrorHandler errorPageErrorHandler) 
```
Change the TODO to say `ErrorHandler` instead of `ErrorProcessor`.

lorban commented on jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletRequestState.java:
```diff
@@ -851,7 +851,7 @@ else if (cause instanceof UnavailableException)
 
         // Set Jetty specific attributes.
-        request.setAttribute(ErrorProcessor.ERROR_EXCEPTION, th);
+        request.setAttribute(org.eclipse.jetty.server.handler.ErrorHandler.ERROR_EXCEPTION, th);
```
There's no need to use the fully qualified class name here.

lorban commented on jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java:
```diff
@@ -493,10 +493,10 @@ public boolean handle()
-                            Request.Processor errorProcessor = ErrorHandler.getErrorProcessor(getServer(), context == null ? null : context.getContextHandler());
+                            Request.Handler errorHandler = ErrorHandler.getErrorHandler(getServer(), context == null ? null : context.getContextHandler());
 
                             // If we can't have a body or have no processor, then create a minimal error response.
```
```suggestion
                            // If we can't have a body or have no handler, then create a minimal error response.
```

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/DelayedHandlerTest.java:
```diff
@@ -463,14 +463,14 @@ public void testNoDelayFormFields() throws Exception
-            public boolean process(Request request, Response response, Callback callback) throws Exception
+            public boolean handle(Request request, Response response, Callback callback) throws Exception
             {
                 // Check that we are called directly from HttpConnection.onFillable via DelayedHandler.process
```
```suggestion
                // Check that we are called directly from HttpConnection.onFillable via DelayedHandler.handle
```

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/HttpChannelTest.java:
```diff
@@ -1331,27 +1331,27 @@ public void testCompletion(List<CompletionTestEvent> events) throws Exception
 
     @ParameterizedTest
     @MethodSource("completionEvents")
     public void testCompletionFailedErrorProcessor(List<CompletionTestEvent> events) throws Exception
```
```suggestion
    public void testCompletionFailedErrorHandler(List<CompletionTestEvent> events) throws Exception
```

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/GracefulHandlerTest.java:
```diff
@@ -409,7 +409,7 @@ public boolean process(Request request, Response response, Callback callback) th
+     * Test for when a Handler returns false from {@link Handler#handle(Request, Response, Callback)}
      * when in graceful mode.
      */
     @Test
```
Rename test to `testHandlerGracefulHandlingFalse()`

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/GracefulHandlerTest.java:
```diff
@@ -355,7 +355,7 @@ public boolean process(Request request, Response response, Callback callback) th
+     * Test for when a Handler returns false from {@link Handler#handle(Request, Response, Callback)}
      * when in normal mode (not during graceful mode).
      * This test exists to ensure that the Callback management of the {@link GracefulHandler} doesn't
      * mess with normal operations of requests.
```
Rename test to `testHandlerNormalHandlingFalse()`

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java:
```diff
@@ -751,9 +751,9 @@ public void testRootReHandlingErrorProcessorLoop() throws Exception
     @Test
     public void testRootReHandlingErrorProcessorExceptionLoop() throws Exception
```
```suggestion
    public void testRootReHandlingErrorHandlerExceptionLoop() throws Exception
```

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java:
```diff
@@ -731,9 +731,9 @@ public void testRootReHandlingErrorProcessor() throws Exception
     @Test
     public void testRootReHandlingErrorProcessorLoop() throws Exception
```
```suggestion
    public void testRootReHandlingErrorHandlerLoop() throws Exception
```

lorban commented on jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java:
```diff
@@ -712,9 +712,9 @@ public boolean process(Request request, Response response, Callback callback)
     @Test
     public void testRootReHandlingErrorProcessor() throws Exception
```
```suggestion
    public void testRootReHandlingErrorHandler() throws Exception
```

lorban commented on jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java:
```diff
@@ -384,12 +384,12 @@ else if (_error.getCause() != x)
+                if (handling)
                 {
                     if (LOG.isDebugEnabled())
                         LOG.debug("already processed, skipping failing callback in {}", HttpChannelState.this);
```
```suggestion
                        LOG.debug("already handled, skipping failing callback in {}", HttpChannelState.this);
```

lorban commented on jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java:
```diff
@@ -487,8 +487,8 @@ public String toString()
             return String.format("%s@%x{processing=%s, processed=%b, writeState=%s, completed=%b, writeCallback=%s, request=%s}",
```
```suggestion
            return String.format("%s@%x{handling=%s, handled=%b, writeState=%s, completed=%b, writeCallback=%s, request=%s}",
```

lorban commented on jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/Rule.java:
```diff
@@ -60,61 +60,61 @@ public String toString()
          * @param response The response
          * @param callback The callback
          * @throws Exception If there is a problem processing
-         * @see #setProcessor(Processor)
```
Change the line above to:
`        * @throws Exception If there is a problem handling`

lorban commented on jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/Rule.java:
```diff
@@ -35,7 +35,7 @@
      * @throws IOException if applying the rule failed
      */
-    public abstract Processor matchAndApply(Processor input) throws IOException;
+    public abstract Handler matchAndApply(Handler input) throws IOException;
```
Change line 41 to:
`    * @return when {@code true}, rules after this one are not handled`

lorban commented on jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/Rule.java:
```diff
@@ -35,7 +35,7 @@
      * @return the possibly wrapped {@code Request} and {@code Processor}, or {@code null} if the rule did not match
      * @throws IOException if applying the rule failed
      */
-    public abstract Processor matchAndApply(Processor input) throws IOException;
```
Change line 35 to:
`     * @return the possibly wrapped {@code Request} and {@code Handler}, or {@code null} if the rule did not match`

Or maybe replace `{@code ...}` with `{@link ...}`?

lorban commented on jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RegexRule.java:
```diff
@@ -69,7 +69,7 @@ public Processor matchAndApply(Processor input) throws IOException
      * @return the possibly wrapped {@code Request} and {@code Processor}
      * @throws IOException if applying the rule failed
      */
-    protected abstract Processor apply(Processor input, Matcher matcher) throws IOException;
```
Change line 69 to:
`     * @return the possibly wrapped {@code Request} and {@code Handler}`

Or use `{@link ...}` instead of `{@code ...}`?

lorban commented on jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/PatternRule.java:
```diff
@@ -59,7 +59,7 @@ public Processor matchAndApply(Processor input) throws IOException
      * @return the possibly wrapped {@code Request} and {@code Processor}
      * @throws IOException if applying the rule failed
      */
-    protected abstract Processor apply(Processor input) throws IOException;
```
Change line 59 to:
`     * @return the possibly wrapped {@code Request} and {@code Handler}`

Or use `{@link ...}` instead of `{@code ...}`?

lorban commented on jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/HeaderRule.java:
```diff
@@ -67,7 +67,7 @@ public Processor matchAndApply(Processor input) throws IOException
      * @return the possibly wrapped {@code Request} and {@code Processor}
      * @throws IOException if applying the rule failed
      */
-    protected abstract Processor apply(Processor input, String value) throws IOException;
```
Change line 67 to:
`     * @return the possibly wrapped {@code Request} and {@code Handler}`

Or use `{@link ...}` instead of `{@code ...}`?

sbordet committed:
Removed Request.__defaultLocale, it was only used internally.

More renamings from *process to *handler for variables, methods, test methods, comments, etc.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

lorban reviewed (changes requested):
You missed a few renames.

We now have `Handler` and `Request.Handler` which adds a bit of confusion, but despite that I like this better.

sbordet committed:
Reverted old-style, obsolete, horrible, soon-to-be-removed (hopefully), almost never used, wrongly named because it's a list and should be plural, `__defaultLocale` field.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
More renaming of methods/variables/comments from process to handle.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>
